### ISSUE TITLE
Macro or tests to ensure storage elements consistency

### ISSUE THREAD

**Issue by mustermeiszer** [centrifuge/centrifuge-chain#912]
Currently, we have a few rather complex storage types. E.g. the permissions storage and the tranches in the pools. My main concern is, that changes to dependent structs might slip through unnoticed and result in runtime errors upon decoding. 

In order to mitigate this risk I would like to either have tests or macros that

* Indicate the memory layout of storage element changed

**Comment by wischli** [comment]
I was wondering whether the following proposal could also be a sufficient solution?

## Proposal D

Add an empty `OnRuntimeUpgrade` migration which just executes the `post_upgrade`  hook guarded by the `try-runtim~. feature to all runtimes. The hook should iterate over all pallets and their storage items and check whether the latter can be decoded.

Then, the execution of the  `try-runtime` hook needs to be added to CI. This would also come in handy for future migrations which would automatically be checked against live chains via CI.

**Comment by mustermeiszer** [comment]
Agreed with you. Parity is also going for your approach https://github.com/paritytech/substrate/issues/12662 ([PR](https://github.com/paritytech/substrate/pull/13013))and they already have a check for migrations in the CI via the `try-state` and fetch state from the live chains. 


* CI needs to check both Altair and Centrifuge states
